mos_tests/nova/nova_tests.py

```diff
@@ -160,3 +169,138 @@ def test_543360_NovaLaunchVMFromVolumeWithAllFlavours(self):
+                                            nics=[{'net-id': net}])
+        self.instances.append(instance.id)
+
+        timeout = time.time() + 60
+        while True:
+            status = self.nova.servers.get(instance).status
+            if status == 'ACTIVE':
+                break
+            elif time.time() > timeout:
+                raise AssertionError("Unable to find running instance")
+            else:
+                time.sleep(1)
+
+        # Assert for attached volumes
+        attached_volumes = self.nova.servers.get(instance).to_dict()[
+            'os-extended-volumes:volumes_attached']
+        self.assertIn({'id': volume.id}, attached_volumes)
+
+        # Assert to flavor size
+        self.assertEqual(self.nova.servers.get(instance).flavor['id'],
+                         initial_flavor,
+                         "Unexpected instance flavor before resize")
+
+        floating_ip = self.nova.floating_ips.create()
+        self.floating_ips.append(floating_ip.ip)
+        instance.add_floating_ip(floating_ip.ip)
+
+        # 3. Resize from m1.small to m1.tiny
+        self.nova.servers.resize(instance, resize_flavor)
+        timeout = time.time() + 60
+        while True:
+            status = self.nova.servers.get(instance).status
+            if status == 'VERIFY_RESIZE':
+                self.nova.servers.confirm_resize(instance)
+                break
+            elif time.time() > timeout:
+                raise AssertionError(
+                    "Unable to find instance in 'VERIFY_RESIZE' state")
+            else:
+                time.sleep(1)
+
+        timeout = time.time() + 120
+        while True:
+            status = self.nova.servers.get(instance).status
+            if status == 'ACTIVE':
+                break
+            elif time.time() > timeout:
+                raise AssertionError("Unable to find running instance")
+            else:
+                time.sleep(1)
+
+        self.assertEqual(self.nova.servers.get(instance).flavor['id'],
+                         resize_flavor,
+                         "Unexpected instance flavor after resize")
+
+        # Check that instance is reachable
+        ping = os.system("ping -c 4 -i 4 {}".format(floating_ip.ip))
+        self.assertEqual(ping, 0, "Instance after resize is not reachable")
+
+    def test_543359_MassivelySpawnVolumes(self):
+        """ This test checks massively spawn volumes
+
+            Steps:
+                1. Create 10 volumes
+                2. Check status of newly created volumes
+                3. Delete all volumes
+        """
+        volume_count = 10
+        volumes = []
+
+        # Creation using Cinder
+        for num in xrange(volume_count):
+            volumes.append(
+                self.cinder.volumes.create(
+                    1, name='Volume_{}'.format(num + 1)))
+            self.volumes.extend(volumes)
+
+        timeout = time.time() + 60
+        while True:
```

Comment:
Could you please implement separate function when will wait while status will be not changed?